src/main/java/zd/zero/waifu/motivator/plugin/WaifuMotivatorProject.java

```diff
@@ -89,7 +90,8 @@ private void checkIfInGoodState( Project projectOpened, Runnable onGoodState ) {
 
     @Override
     public void projectClosing( @NotNull Project project ) {
-        if ( !pluginState.isSayonaraEnabled() || isMultipleProjectsOpened() ) return;
```

Comment:
I was curious, how come the plugin state was saved during initialization? When state changes, this will become outdated. It's not an issue at the moment, I was just wondering.

Reply:
> how come the plugin state was saved during initialization? 

Can you provide more information?

Reply:
Sure! 
I was wondering why this was done
`this.pluginState = WaifuMotivatorPluginState.getPluginState();`

instead of just using ` WaifuMotivatorPluginState.getPluginState()` when state was needed. 

However, I think, I may be thinking incorrectly. Maybe the reference to the `WaifuMotivatorState ` object is the same one that gets changed as well. So maybe it is not a big deal to save the reference in an instance field.

Reply:
> I was wondering why this was done

😅😅😅
I got lazy on keep calling `WaifuMotivatorPluginState.getPluginState()` so I made it as a class member.



> Maybe the reference to the WaifuMotivatorState object is the same one that gets changed as well.

You are correct as they are the same object reference pointing to the application service of the plugin state as it's a singleton.
```
ApplicationManager.getApplication().getService( WaifuMotivatorPluginState.class );
```



Now that I'm thinking about it it's unreadable to pass it as a class member because it loses its intent, so what I'm going to do is remove the `pluginState` member and always get a reference to the plugin state by always calling `WaifuMotivatorPluginState.getPluginState()`